internal/databases/postgres/daemon_handler.go

```diff
@@ -3,13 +3,64 @@ package postgres
 import (
 	"bytes"
 	"encoding/binary"
+	"errors"
+	"fmt"
 	"net"
 	"os"
 	"path"
 
 	"github.com/wal-g/tracelog"
 )
 
+type CheckType byte
+type FileType byte
+
+type MessageSocketHandler interface {
+	Handle(messageBody []byte, c net.Conn, f func(string) error) error
+}
+
+func MessageTypeConstruct(messageType byte) MessageSocketHandler {
+	switch messageType {
+	case 'C':
+		return CheckType('C')
+	case 'F':
+		return FileType('F')
+	default:
+		return nil
+	}
+}
+
+func (msg CheckType) Handle(messageBody []byte, c net.Conn, f func(string) error) error {
```

Comment:
You need to create a separate message handler structure for each message type.

i.e.
```go
type SocketMessageHandler interface {
Handle(messageBody []byte) error
}


type ArchiveMessageHandler struct
{
...
}

func (h * ArchiveMessageHandler) Handle(messageBody []byte) error {}

type CheckMessageHandler struct
{
...
}

func (h * CheckMessageHandler) Handle(messageBody []byte) error {}
```